bitrepository-core/src/main/java/org/bitrepository/common/utils/SettingsUtils.java

```diff
@@ -101,41 +101,51 @@ public static String getCollectionName(String collectionID) {
         return name;
     }
 
+    public static IntegrityServiceSettings getIntegrityServiceSettings() {
+        return settings.getReferenceSettings().getIntegrityServiceSettings();
+    }
+
+    /**
+     * Grab all PillarDetails from the ReferenceSettings if any are defined and return them as a list.
+     * Return an empty list if not defined.
+     * @return List of all PillarDetails defined in the ReferenceSettings
+     */
+    public static List<PillarIntegrityDetails.PillarDetails> getPillarIntegrityDetails() {
+        PillarIntegrityDetails pillarIntegrityDetails = getIntegrityServiceSettings().getPillarIntegrityDetails();
+        if (pillarIntegrityDetails != null) {
+            List<PillarIntegrityDetails.PillarDetails> pillarDetails = pillarIntegrityDetails.getPillarDetails();
+            if (pillarDetails != null) {
+                return pillarDetails;
```

Comment:
`getPillarDetails()` returns a mutable list that we in turn return. This allows the caller to modify the list held in the pillar integrity details. Which we don’t want. We should wrap in `Collections.unmodifiableList()` either in `getPillarDetails()` or here or both.

Reply:
Good point!